### PR TITLE
BHBC-1555

### PR DESCRIPTION
**Pull request by cgarrettjones**
# Overview

## Links to jira tickets

- [BHBC-1555](https://quartech.atlassian.net/browse/BHBC-1555)

## This PR contains the following changes

- first cut of generic project management database

## This PR contains the following types of changes

- [ ] New feature (change which adds functionality)
- [ ] Enhancement (improvements to existing functionality)
- [ ] Bug fix (change which fixes an issue)
- [ ] Misc cleanup / Refactoring / Documentation
- [ ] Version change

## Checklist

A list of items that are good to consider when making any changes.

_Note: this list is not exhaustive, and not all items are always applicable._

### General

- [ ] I have performed a self-review of my own code

### Code

- [ ] New files/classes/functions have appropriately descriptive names and comment blocks to describe their use/behaviour
- [ ] I have avoided duplicating code when possible, moving re-usable pieces into functions
- [ ] I have avoided hard-coding values where possible and moved any re-usable constants to a constants file
- [ ] My code is as flat as possible (avoids deeply nested if/else blocks, promise chains, etc)
- [ ] My code changes account for null/undefined values and handle errors appropriately
- [ ] My code uses types/interfaces to help describe values/parameters/etc, help ensure type safety, and improve readability

### Style

- [ ] My code follows the established style conventions
- [ ] My code uses native material-ui components/icons/conventions when possible

### Documentation

- [ ] I have commented my code sufficiently, such that an unfamiliar developer could understand my code
- [ ] I have added/updated README's and related documentation, as needed

### Tests

- [ ] I have added/updated unit tests for any code I've added/updated
- [ ] I have added/updated the Postman requests/tests to account for any API endpoints I've added/updated

### Linting/Formatting

- [ ] I have run the linter and fixed any issues, as needed  
       _See the `lint` commands in package.json_
- [ ] I have run the formatter and fixed any issues, as needed  
       _See the `format` commands in package.json_

### SonarCloud

- [ ] I have addressed all SonarCloud Bugs, Vulnerabilities, Security Hotspots, and Code Smells

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

## Screenshots

Please add any relevant UI screenshots, if applicable.

### CODE REVIEW

**Comment by NickPhura** [comment]
There are a whole bunch of references to `biohub` that should probably be changed to something else. Looks like Roland was using `restoration` as the new name for things, which sounds fine.

Some of the file names also reference biohub.

**Review by rstens — COMMENTED** [review]
Yes the changes that i made in the titles and system name are as follows:
biohubbc -> restoration-tracker
biohub -> restoration

**Comment by cgarrettjones** [comment]
are we making a generic copy? naming wouldn't be an issue. either way...

**Comment by NickPhura** [comment]
> are we making a generic copy? naming wouldn't be an issue. either way...

Once we have the api and app parts merged in, we can clone the repo to maintain a copy that stays with a more minimal working setup, for future use.  

Our original repo already has a mixture of `sims` and `biohub` naming, so in this repo we should at least unify all of the naming now while its easy to do. Obviously the schemas could be named anything and still work, but I think its nice if they all match, so future devs don't get confused, etc.

**Review by rstens — APPROVED** [review]

File: database/src/migrations/20210715170005_adding_system_rules.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+  set schema '${DB_SCHEMA}';
+  set search_path = ${DB_SCHEMA},public;
+
+  -- Add system rules
+  insert into  security_rule (name, rule_definition, system_rule, record_effective_date) values ('New Project Attachment','{"target":"project_attachment","rule":"","project":""}', true, now()); -- 1
+  insert into  security_rule (name, rule_definition, system_rule, record_effective_date) values ('New Survey Attachment','{"target":"survey_attachment","rule":"","project":""}', true, now()); -- 2
+  insert into  security_rule (name, rule_definition, system_rule, record_effective_date) values ('New Occurrence','{"target":"occurrence","rule":"","project":""}', true, now()); -- 3
+  insert into  security_rule (name, rule_definition, system_rule, record_effective_date) values ('New Project Report Attachment','{"target":"project_report_attachment","rule":"","project":""}', true, now()); -- 4
+  insert into  security_rule (name, rule_definition, system_rule, record_effective_date) values ('New Survey Report Attachment','{"target":"survey_report_attachment","rule":"","project":""}', true, now()); -- 5
```
Comment (anissa-agahchen) [inline database/src/migrations/20210715170005_adding_system_rules.ts:16]: can you remove the survey_report_attachment?  We're not keeping anything related to surveys in the app and api.

**Pull request merged**